test/TokenServiceUpgradeFlow.js

```diff
@@ -0,0 +1,203 @@
+                signers.map((signer) => signer.address),
+                threshold,
+            )
+            .then((d) => d.deployed());
+
+        service = await deployInterchainTokenService(
+            wallet,
+            create3Deployer.address,
+            tokenManagerDeployer.address,
+            standardizedTokenDeployer.address,
+            gateway.address,
+            gasService.address,
+            remoteAddressValidator.address,
+            tokenManagerImplementations.map((impl) => impl.address),
+            deploymentKey,
+            axelarServiceGovernance.address,
+        );
+    });
+
+    it('should upgrade Interchain Token Service through AxelarServiceGovernance timeLock proposal', async () => {
+        const commandID = 0;
+        const target = service.address;
+        const nativeValue = 0;
+        const timeDelay = isHardhat ? 12 * 60 * 60 : 12;
+
+        const targetInterface = new Interface(service.interface.fragments);
+        const newServiceImplementation = await deployContract(wallet, 'InterchainTokenService', [
+            tokenManagerDeployer.address,
+            standardizedTokenDeployer.address,
+            gateway.address,
+            gasService.address,
+            remoteAddressValidator.address,
+            tokenManagerImplementations.map((impl) => impl.address),
+        ]);
+        const newServiceImplementationCodeHash = await getBytecodeHash(newServiceImplementation);
+        const setupParams = '0x';
+        const calldata = targetInterface.encodeFunctionData('upgrade', [
+            newServiceImplementation.address,
+            newServiceImplementationCodeHash,
+            setupParams,
+        ]);
+
+        const [payload, proposalHash, eta] = await getPayloadAndProposalHash(commandID, target, nativeValue, calldata, timeDelay);
+
+        const commandIdGateway = getRandomBytes32();
+        const sourceTxHash = keccak256('0x123abc123abc');
+        const sourceEventIndex = 17;
+
+        await approveContractCall(
+            gateway,
+            governanceChain,
+            governanceAddress,
+            axelarServiceGovernance.address,
+            payload,
+            sourceTxHash,
+            sourceEventIndex,
+            commandIdGateway,
+        );
+
+        const txExecute = await axelarServiceGovernance.execute(commandIdGateway, governanceChain, governanceAddress, payload);
+
+        const receiptExecute = await txExecute.wait();
+        const minimumEta = (await ethers.provider.getBlock(receiptExecute.blockNumber)).timestamp + buffer;
+        const finalEta = minimumEta > eta ? minimumEta : eta;
+
+        await expect(txExecute)
+            .to.emit(axelarServiceGovernance, 'ProposalScheduled')
+            .withArgs(proposalHash, target, calldata, nativeValue, finalEta);
+
+        await waitFor(timeDelay);
+
+        const tx = await axelarServiceGovernance.executeProposal(target, calldata, nativeValue);
+        const receipt = await tx.wait();
```

Comment:
can you test this on live network? add confirmations to the wait/deployment txs

Reply:
yup, it's passing on live networks